src/nft.sol

```
@@ -13,185 +13,156 @@
-      * @return result the converted address
-      */
-    function bytesToUint(bytes memory data) internal pure returns (uint) {
-        require(data.length <= 256, "slicing out of range");
-        return abi.decode(data, (uint));
-     }
-
-     /**
-      * @dev Parses a uint and returns the hex string
-      * @param payload uint
-      * @return string the corresponding hex string
-      */
-     function uintToHexStr(uint payload) internal pure returns (string memory) {
-       if (payload == 0)
-         return "0";
-       // calculate string length
-       uint i = payload;
-       uint length;
-
-       while (i != 0) {
-         length++;
-         i = i >> 4;
-       }
-       // parse byte by byte and construct the string
-       i = payload;
-       uint mask = 15;
-       bytes memory result = new bytes(length);
-       uint k = length - 1;
-
-       while (i != 0) {
-         uint curr = (i & mask);
-         result[k--] = curr > 9 ? byte(55 + uint8(curr)) : byte(48 + uint8(curr));
-         i = i >> 4;
-       }
-
-       return string(result);
-     }
-
-    function equalBytes(bytes memory a, bytes memory b) internal pure returns (bool) {
-        if (a.length == b.length) {
-            for (uint i = 0; i < a.length; i++) {
-                if (a[i] != b[i]) {
-                    return false;
-                }
-            }
-            return true;
+    /**
+     * Computes the asset hash from the given params and verifies if its present in the AssetManager
+     * @param to address deposit address of the token
+     * @param properties bytes[] properties of the each proof
+     * @param values bytes[] value associated with each property
+     * @param salts bytes[] salt associated with each property
+     */
+    function _verifyAsset(address to, bytes[] memory properties, bytes[] memory values, bytes32[] memory salts) internal view returns (bool) {
+        require(to != address(0), "not a valid address");
```

I don't think we need to catch this use case on chain. It will fail at a later point anyway.

---

I've reopened this until it is addressed in the PR

---

always fail-fast :)

---

I have removed this check for now, but I would love it be there

---

fail fast for extra gas is not worth it IMO. 99.999% of times this will be not be triggered but it will still cost people gas every time.